### PR TITLE
Catch2: handle exceptions

### PR DESCRIPTION
**Pull request by jnewb1**
Exceptions are a separate output in the XML from normal failures

### CODE REVIEW

File: tests/test_pytest_cpp.py [hunk 1]
```diff
@@ -628,6 +628,10 @@ def test_catch2_failure(exes):
         )
         assert len(failures) == 2
 
+        # test exceptions
+        failures, _ = facade.run_test(exes.get(f"catch2_error{suffix}"), "Error")
+        assert len(failures) == 1
```
Comment (nicoddemus) [inline tests/test_pytest_cpp.py:633]: Should we test that the failure contains the `a runtime error` message?

**Review by nicoddemus — APPROVED** [review]

File: tests/test_pytest_cpp.py (hunk 1 above)
Comment (jnewb1) [reply tests/test_pytest_cpp.py:633]: done!

File: tests/test_pytest_cpp.py [hunk 2]
```diff
@@ -628,6 +628,14 @@ def test_catch2_failure(exes):
...
+
+        fail1 = failures[0]
```
Comment (nicoddemus) [inline tests/test_pytest_cpp.py:635]: Minor, this does the same but is more compact:

```suggestion
        [fail1] = failures
```

**Comment by nicoddemus** [comment]
Thanks @jnewb1, great work!

Do you have more Catch2 work in the horizon? If not, feel free to kick off a new release (see [RELEASING.md.](https://github.com/pytest-dev/pytest-cpp/blob/master/RELEASING.rst)).

**Comment by jnewb1** [comment]
> Thanks @jnewb1, great work!
> 
> 
> 
> Do you have more Catch2 work in the horizon? If not, feel free to kick off a new release (see [RELEASING.md.](https://github.com/pytest-dev/pytest-cpp/blob/master/RELEASING.rst)). 

I'm mostly fixing them as I run into them in openpilot. I don't think it handles when the test crashes and doesn't make the xml. Going to fix that, then we can cut the release

**Pull request merged**